packages/admin-panel-server/src/viz-builder/dashboardVisualisation/DashboardVisualisationExtractor.ts

```diff
@@ -62,15 +72,14 @@ export class DashboardVisualisationExtractor {
     };
   }
 
-  getDashboardItem() {
+  public getDashboardItem(): ExpandType<yup.InferType<DashboardItemValidator>> {
```

Comment:
I believe we don't need to infer types here either, since `yup.validateSync` takes care of constructing a correct type for the result:

```suggestion
  public getDashboardItem() {
```

Similary for `getReport`

Reply:
Yeah that's true! However, I wanted to wrap the whole in the `ExpandType` type so that the output type is nice and human readable (otherwise it's this crazy Yup generic type thing). So I need to pass in the return type to the `ExpandType`... I'll take a look and see if I can get that working without explicitly casting the type tho 👍

Reply:
I see what you mean! You are right that the return type is not very human-friendly... In [ mel-454-viz-builder-export-improvements - e265ee3](https://github.com/beyondessential/tupaia/commit/e265ee30b9724597d4532a22a9b4bad139f542bc) I added a [DashboardItemRecord](https://github.com/beyondessential/tupaia/blob/dc12fcc943e292dc794068c0288ad18dfebe1a69/packages/admin-panel-server/src/viz-builder/types.ts#L67) record, so we can also use that for clarity - and type matching.

Happy to leave this as is and revisit when that branch is reviewed!